### PR TITLE
Fix the case that labels are all ignored

### PR DESCRIPTION
**Pull request by lyakaap**
An error occurred in line 177 ( `C = probas.size(1)` ) when input labels are all ignored, because labels got empty.
This PR deal with such problems.

One thing I concerned about is the difference between the lines I added and below lines
```
        if len(labels) == 0:
            # only void pixels, the gradients should be 0
            return logits.sum() * 0.
```

Above is a part of the original implementation of lovasz_flat. This doesn't exclude void labels for calculation the loss. Is it excepted behavior?

### CODE REVIEW

**Comment by bermanmaxim** [comment]
Hi,
Thanks for the PR. However, I think it is better to use `logits.sum() * 0.` rather than `NaN` since you want the loss to be `0.`.
Since the gradient of `logits` w.r.t. `logits.sum() * 0.` is `0.` it doesn't matter if you exclude void labels or not. Said differently, I needed something that gave the `0.` value and had gradient `0.`. (I could also have simply created a new `0.` tensor for that on the right device).
I would rather change `line 177` into
```
if probas.numel() == 0:
    return probas * 0.
C = probas.size(1)
```
If you agree and make these changes (instead of resorting to `nan`) I will merge.

**Comment by lyakaap** [comment]
Thank you for the explanation. I updated code as you said.

**Comment by bermanmaxim** [comment]
Thanks for noticing the small bug with all-void inputs.
